tests/custom_node.rs

```diff
@@ -0,0 +1,78 @@
+use quote::{ToTokens, TokenStreamExt};
+use rstml::{
+    atoms::{self, OpenTag, OpenTagEnd},
+    node::{CustomNode, Node, NodeElement},
+    recoverable::Recoverable,
+};
+use syn::{parse_quote, Expr, Token};
+
+#[derive(Debug, syn_derive::ToTokens)]
+struct If {
+    token_lt: Token![<],
+    token_if: Token![if],
+    condition: Expr,
+    open_tag_end: OpenTagEnd,
+    #[to_tokens(TokenStreamExt::append_all)]
+    body: Vec<Node>,
+    close_tag: Option<atoms::CloseTag>,
+}
+
+impl CustomNode for If {
+    fn to_tokens(&self, tokens: &mut proc_macro2::TokenStream) {
+        ToTokens::to_tokens(&self, tokens)
+    }
+
+    fn peek_element(input: syn::parse::ParseStream) -> bool {
+        input.peek(Token![<]) && input.peek2(Token![if])
+    }
+
+    fn parse_element(
+        parser: &mut rstml::recoverable::RecoverableContext,
+        input: syn::parse::ParseStream,
+    ) -> Option<Self> {
+        let token_lt = OpenTag::parse_start_tag(parser, input)?;
+        let token_if = parser.parse_simple(input)?;
+        let (condition, open_tag_end): (_, OpenTagEnd) = parser.parse_simple_with_ending(input)?;
+        let (body, close_tag) = if open_tag_end.token_solidus.is_none() {
+            // Passed to allow parsing of close_tag
+            NodeElement::parse_children(
+                parser,
+                input,
+                false,
+                &OpenTag {
+                    token_lt,
+                    name: parse_quote!(#token_if),
+                    generics: Default::default(),
+                    attributes: Default::default(),
+                    end_tag: open_tag_end.clone(),
+                },
+            )?
+        } else {
+            (Vec::new(), None)
+        };
+        Some(Self {
+            token_lt,
+            token_if,
+            condition,
+            open_tag_end,
+            body,
+            close_tag,
+        })
+    }
+}
```

Comment:
I made some of the functionality used by `Node`'s `ParseRecoverable` implementation public, so this is easier.

But I'm not sure if this is the right way to do so.

Reply:
If you feel like it makes more sense to only merge the `CustomNode` for now and skip on exposing more parsing helpers, I'd be fine with that as well.

Reply:
It's okay to make them public for now, but currently i have no plan to keep them stable, and i think this should be documented.
Most of this method was done specifically for the task and i don't find them reusable.

For example, using `parse_simple_with_ending` in `If` prevent user from using `>` (great sign, compare operator) in condition, but allows to use `<` which i find controversial.